File: AdvantageFramework.Core/Database/Entities/AlertCategory.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("ALERT_CATEGORY")]
    public partial class AlertCategory
    {
        public AlertCategory()
        {
            AlertGroups = new HashSet<AlertGroup>();
            AlertStates = new HashSet<AlertState>();
            Alerts = new HashSet<Alert>();
        }

        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        [Key]
        [Column("ALERT_CAT_ID")]
        public int AlertCatId { get; set; }
        [Column("ALERT_TYPE_ID")]
        public int AlertTypeId { get; set; }
        [Required]
        [Column("ALERT_DESC")]
        [StringLength(40)]
        public string AlertDesc { get; set; }
        [Column("PROMPT")]
        public short? Prompt { get; set; }
        [Column("GROUP_LVL_SECURITY")]
        public short? GroupLvlSecurity { get; set; }
        [Column("PDF_ATTACHMENT")]
        public short? PdfAttachment { get; set; }
        [Column("IS_USER_DEFINED")]
        public bool? IsUserDefined { get; set; }
        [Column("IS_INACTIVE")]
        public bool? IsInactive { get; set; }
        [Column("COLOR_CODE")]
        [StringLength(25)]
        public string ColorCode { get; set; }
        [Column("BOARD_TMPLT_COL_ID")]
        public int? BoardTmpltColId { get; set; }
        [Column("PARENT_CAT_ID")]
        public int? ParentCatId { get; set; }

        [ForeignKey(nameof(AlertTypeId))]
        [InverseProperty("AlertCategories")]
        public virtual AlertType AlertType { get; set; }
        [InverseProperty(nameof(AlertGroup.AlertCat))]
        public virtual ICollection<AlertGroup> AlertGroups { get; set; }
        [InverseProperty(nameof(AlertState.DfltAlertCat))]
        public virtual ICollection<AlertState> AlertStates { get; set; }
        [InverseProperty(nameof(Alert.AlertCat))]
        public virtual ICollection<Alert> Alerts { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/AlertCategory.vb
Namespace Database.Entities

	<Table("ALERT_CATEGORY")>
	Public Class AlertCategory
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties

            ID
            AlertTypeID
            Description
            AllowPrompt
            GroupLevelSecurityID
            HasPDFAttachment
            IsUserDefined
            IsInactive

            Alerts
            AlertType
            AlertGroupCategories
            AlertStates

        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
		<Required>
        <Column("ALERT_CAT_ID")>
        Public Property ID() As Integer
        <Column("ALERT_TYPE_ID")>
        Public Property AlertTypeID() As Nullable(Of Integer)
        <Required>
		<MaxLength(40)>
		<Column("ALERT_DESC", TypeName:="varchar")>
		Public Property Description() As String
		<Column("PROMPT")>
		Public Property AllowPrompt() As Nullable(Of Short)
		<Column("GROUP_LVL_SECURITY")>
		Public Property GroupLevelSecurityID() As Nullable(Of Short)
        <Column("PDF_ATTACHMENT")>
        Public Property HasPDFAttachment() As Nullable(Of Short)
        <Column("IS_USER_DEFINED")>
        Public Property IsUserDefined As Nullable(Of Boolean) = False
        <Column("IS_INACTIVE")>
        Public Property IsInactive As Nullable(Of Boolean) = False

        Public Overridable Property Alerts As ICollection(Of Database.Entities.Alert)
        Public Overridable Property AlertType As Database.Entities.AlertType
        Public Overridable Property AlertGroupCategories As ICollection(Of Database.Entities.AlertGroupCategory)
        Public Overridable Property AlertStates As ICollection(Of Database.Entities.AlertState)

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID

        End Function

#End Region

	End Class

End Namespace
